Adding Cmdlet New-AzApplicationGatewayRewriteRuleUrlConfiguration to support url rewrite capability for V2 application gateways

abjai opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->
This should be released as a part of Powershell January Release

Adding Support for UrlConfiguration under RewriteRuleActionset for Application Gateway Url Rewrite Feature

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

Powershell design review link: Azure/azure-powershell-cmdlet-review-pr#339
Swagger PR: Azure/azure-rest-api-specs#7905

abjai commented on src/Network/Network.Test/ScenarioTests/ApplicationGatewayTests.ps1:
```diff
@@ -973,6 +973,262 @@ function Test-ApplicationGatewayCRUDRewriteRuleSetWithConditions
 	}
 }
 
+function Test-ApplicationGatewayCRUDRewriteRuleSetWithUrlConfiguration
```
Playback can not be done since the manifest for 2019-01-11 is not updated

isra-fel replied:
When will it be available?

abjai replied:
Added

abjai replied:
added

isra-fel commented on src/Network/Network/help/New-AzApplicationGatewayRewriteRuleUrlConfiguration.md:
````diff
@@ -0,0 +1,117 @@
+
+### Example 1
+```powershell
+PS C:\> $urlCongiguration = New-AzApplicationGatewayRewriteRuleUrlConfiguration -ModifiedPath "/abc" -ModifiedQueryString "x=y&a=b"
````
```suggestion
PS C:\> $urlConfiguration = New-AzApplicationGatewayRewriteRuleUrlConfiguration -ModifiedPath "/abc" -ModifiedQueryString "x=y&a=b"
```

isra-fel commented on src/Network/Network/help/New-AzApplicationGatewayRewriteRuleUrlConfiguration.md:
````diff
@@ -0,0 +1,117 @@
+PS C:\> $urlCongiguration = New-AzApplicationGatewayRewriteRuleUrlConfiguration -ModifiedPath "/abc" -ModifiedQueryString "x=y&a=b"
+```
+
+This command creates a rewrite rule url configuration and stores the result in the variable named $urlCongiguration.
````
```suggestion
This command creates a rewrite rule url configuration and stores the result in the variable named $urlConfiguration.
```

isra-fel commented:
/azp run

abjai commented on src/Network/Network/Network.csproj:
```diff
@@ -12,7 +12,7 @@
   <ItemGroup>
     <PackageReference Include="AutoMapper" Version="6.2.2" />
-    <PackageReference Include="Microsoft.Azure.Management.Network" Version="19.17.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Network" Version="21.0.0-preview" />
```
This will be available once swagger changes as published

abjai commented:
> Hi @abjai, please see my inline comments. Thanks!
> Oh and please also fix the merge conflict.

@isra-fel I have addressed your comments. I have added the local feed so that tests can run. Once swagger SDK got published and updated in december-release branch, i will pull those changes and resolve the conflicts.

Please note that we are targetting these changes to be released as a part of January end release.

isra-fel commented:
I'm good with the change. However the CI is blocked by the merge conflict.

abjai commented:
> Hi @abjai
> I'm good with the change. However the CI is blocked by the merge conflict.

@isra-fel : I have removed the conflict changes however i could see that SDK is still not updated in Network.csproj so tests might fail.

@number213 : Can you please update the network.csproj with latest SDK so that tests start passing?

Please note that these changes need to be part of Januray 20 release

isra-fel commented:
So, your test project still relies on 21.0.00preview SDK.

abjai commented:
> So, your test project still relies on 21.0.00preview SDK.

@isra-fel : Yes it is still dependent on the updated SDK published as a part of January 17 release. I don't know the exact version published as a part of SDK release. Adding @number213 to help in updating the SDK version for Network.csproj

abjai commented:
> > So, your test project still relies on 21.0.00preview SDK.
> 
> @isra-fel : Yes it is still dependent on the updated SDK published as a part of January 17 release. I don't know the exact version published as a part of SDK release. Adding @number213 to help in updating the SDK version for Network.csproj

@isra-fel @number213 has raised a PR to update SDK https://github.com/Azure/azure-powershell/pull/10928. Once it is merged, my changes should be able to pass the tests

abjai commented:
> > > So, your test project still relies on 21.0.00preview SDK.
> > 
> > 
> > @isra-fel : Yes it is still dependent on the updated SDK published as a part of January 17 release. I don't know the exact version published as a part of SDK release. Adding @number213 to help in updating the SDK version for Network.csproj
> 
> @isra-fel @number213 has raised a PR to update SDK #10928. Once it is merged, my changes should be able to pass the tests

@isra-fel : updated the PR with latest sdk.

isra-fel commented:
/azp run azure-powershell - powershell-core

isra-fel commented:
/azp run azure-powershell - powershell-core

number213 commented:
Need to work on PR for merging network into master. Since @isra-fel approved and added "waiting for CI" I'll go ahead and merge this PR
